### PR TITLE
 Fix #12 Add a test for "unset" value

### LINKED ISSUE #12: Add a test for "unset" value
Citing from [1]:

> For any property, a value of `unset` is to remove the effect of that property, even if it has been set before.

Searching for `unset` in the core-test project returns zero matches.

Expected: there should be at least one test for each of the widely accepted properties that unsets a previously set value. This is i.a. important to make sure that handlers for properties that look like boolean or integer at the first sight can handle `unset` too.

[1] https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties#current-universal-properties

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

**Comment by ppalaga** [comment]
@xuhdev could you please review? I checked that the new tests pass with both current `core-c` and current `core-py`.

**Comment by ppalaga** [comment]
@xuhdev as I said in #12 , this just codifies the current behavior of `editorconfig-core-py` and `editorconfig-core-c`. It would be nice to have it in the suite so that other implementations can makes sure that they comply too.

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,6 @@ install_manifest.txt
 # Generated by CTest
 /Testing
 CTestTestfile.cmake
+
+# Eclipse
+.project
```
Comment (xuhdev) [inline .gitignore]: Please keep the new line char.

File: filetree/CMakeLists.txt [hunk 2]
```diff
@@ -1,16 +1,16 @@
 #
 # Copyright (c) 2011-2012 EditorConfig Team
 # All rights reserved.
-#
```
Comment (xuhdev) [inline filetree/CMakeLists.txt]: Please make these trailing whitespace removal into a separate commit.

File: .gitignore (hunk 1 above)
Comment (ppalaga) [reply .gitignore]: Fixed in 9b6999c

File: filetree/CMakeLists.txt (hunk 2 above)
Comment (ppalaga) [reply filetree/CMakeLists.txt]: Sorry, I just noticed you have an `.editorconfig` file saying `trim_trailing_whitespace = false`. Hence I added that whitespace back in 861284f

**Comment by ppalaga** [comment]
@xuhdev thanks for the review, all your requests should be fixed in 
9b6999c and 861284f .

**Comment by xuhdev** [comment]
Thanks!

**Comment by ppalaga** [comment]
Thanks for merging @xuhdev!
